src/unistd.rs

```diff
@@ -11,18 +11,126 @@ use std::os::unix::io::RawFd;
 use std::path::{PathBuf};
 use void::Void;
 use sys::stat::Mode;
+use std::fmt;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 pub use self::linux::*;
 
+/// User identifier
+///
+/// Newtype pattern around `uid_t` (which is just alias). It prevents bugs caused by accidentally
+/// passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct Uid(uid_t);
+
+impl Uid {
+    pub fn from_raw(uid: uid_t) -> Self {
+        Uid(uid)
+    }
+
+    pub fn current() -> Self {
+        getuid()
+    }
+
+    pub fn effective() -> Self {
+        geteuid()
+    }
+
+    pub fn is_root(self) -> bool {
```

Comment:
In this case, you're right that `is_root(self)` and `is_root(&self)` are probably equivalent.  But stylistically, the latter is preferred.  In the entire Nix crate, we only use the former style in one place, and that's because a std trait requires it.  The nonconsuming variety is also more future-proof.  For example, we might someday turn these methods into a Trait, and implement it for other objects that are harder to clone, like SIDs.  Are there any downsides to taking `self` by reference?